src/ui/React/LinearForwardProgress.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from "react";
+import { useRef } from "react";
+import LinearProgress from "@mui/material/LinearProgress";
+
+interface LinearForwardProgressProps {
+  value: number;
+  backgroundColor?: React.CSSProperties["color"];
+  barColor?: React.CSSProperties["color"];
+}
+
+// Round percentage to nearest interval to limit keyframe definitions
+const roundToInterval = (value: number, interval = 10): number => {
+  return Math.round(value / interval) * interval;
+};
+
+// Generate predefined keyframes for forward animations
+const generateKeyframes = (startPercent: number, endPercent: number) => {
+  const startPos = startPercent - 100;
+  const endPos = endPercent - 100;
+
+  return {
+    "0%": { transform: `translateX(${startPos}%)` },
+    "50%": { transform: "translateX(0%)" },
```

Comment:
I just realized, you changed how this is working and the animation doesn't have constant speed anymore. It always spends half the time getting too the end and half the time coming from the beginning.